src/com/projectkorra/projectkorra/firebending/util/FireDamageTimer.java

```diff
@@ -15,17 +16,35 @@
 public class FireDamageTimer {
 
 	private static final int MAX_TICKS = 90;
+	private static Ability ability = null;
 	private static final int DAMAGE = 1;
 	private static final long BUFFER = 30;
 	private static final Map<Entity, Player> INSTANCES = new ConcurrentHashMap<>();
 	private static final Map<Entity, Long> TIMES = new ConcurrentHashMap<>();
 
+	/**
+	 * Deprecated. Use FireDamageTimer#FireDamageTimer(final Entity entity, Final Player source, Ability abil)
+	 * instead.
+	 * @param entity affected entity.
+	 * @param source player who used the fire move.
+	 */
+	@Deprecated
 	public FireDamageTimer(final Entity entity, final Player source) {
 		if (entity.getEntityId() == source.getEntityId()) {
 			return;
 		}
 
 		INSTANCES.put(entity, source);
+		ability = null;
```

Comment:
I think that you should instead call `this(entity, source, null);` rather than have another clone of the constructor

Reply:
I like this!